### ISSUE TITLE
Textinput value not updating

### ISSUE THREAD

**Issue by AnkS4** [beeware/beeware#160]
**Describe the bug**
Text is returned in the terminal instead of App window.

**To Reproduce**
1. Define Text Input & Box
box = toga.Box(style=Pack(direction=COLUMN, padding=5, flex=1))
self.output = toga.TextInput(style=Pack(flex=1))
2. Add TextInput object to the box
box.add(self.output)
3. Add box to the main box
main_box.add(box)
4. On Button Click invoke say_hello, set self.output.value to "Hello"
def say_hello(self, widget):
        print("Hello!", self.output.value)
5. "Hello" is printed in terminal instead of box with TextInput object.

**Environment:**
 - Operating System: Manjaro, GNU/Linux
 - Python version: Python 3.10.4 / 3.8.5
 - Software versions:
   - Briefcase: 0.3.8
   - Toga: 0.3.0.dev34

**Terminal Output**
```
$ briefcase dev

[speechtotext] Starting in dev mode...
[GTK+] Not implemented: TextInput.set_alignment()
[GTK+] Not implemented: TextInput.clear_error()
[GTK+] Not implemented: TextInput.set_on_lose_focus()
[GTK+] Not implemented: TextInput.set_on_gain_focus()
/home/anks/beeware/beeware-venv/lib/python3.10/site-packages/toga_gtk/app.py:40: DeprecationWarning: Gtk.Window.set_wmclass is deprecated
  self.native.set_wmclass(app.interface.name, app.interface.name)
Hello

```

Edit: Also, reproducible in Python 3.8.5

**Comment by freakboy3742** [comment]
Thanks for the report. This is a known problem. The underlying problem appears to be that GTK/Gnome has deprecated one API without providing an alternative that works.

The issue is the app presenting "python" or "__main__.py" in the Application menu bar, rather than the app's formal name (Not the title bar of the window - the top-of-screen menu bar. This *might* be an Ubuntu-specific issue)

At least for now, it's only a warning, not an error, so it can be safely ignored. However, we'd love to apply a fix if someone can find one.

**Comment by mhsmith** [comment]
> * On Button Click invoke say_hello, set self.output.value to "Hello"
>   def say_hello(self, widget):
>   print("Hello!", self.output.value)
> * "Hello" is printed in terminal instead of box with TextInput object.

This isn't related to the GTK warning, but the right way of updating a TextInput value is:
```
self.output.value = "Hello!"
```

**Comment by freakboy3742** [comment]
I clearly skimmed the body of this ticket much too quickly. I saw the title and assumed the body was instructions for reproducing the message from the subject in the ticket. On a second read, I see that the actual problem is about updating the value in the text box. 

I've updated the subject of the ticket. Closing on the basis that it's an error of usage; the solution provided by @mhsmith is the correct usage.
